pkg/yang/yang.go

```diff
@@ -179,17 +179,26 @@ func (s *Module) FullName() string {
 // GetPrefix returns the proper prefix of m.  Useful when looking up types
 // in modules found by FindModuleByPrefix.
 func (m *Module) GetPrefix() string {
-	if m == nil {
+	pfx := m.getPrefix()
+	if pfx == nil {
 		// This case can be true during testing.
 		return ""
 	}
-	if m.Prefix != nil {
-		return m.Prefix.Name
+	return pfx.Name
+}
+
+func (m *Module) getPrefix() *Value {
+	if m == nil {
+		return nil
+	} else if m.Prefix != nil {
+		return m.Prefix
+	} else if m.BelongsTo != nil {
+		return m.BelongsTo.Prefix
 	}
```

Comment:
This should be written as a switch statement:

switch {
case m == nil:
    return nil
case m.Prefix != nil:
    return m.Prefix
case m.BelongsTo != nil:
    return m.BelongsTo.Prefix
default:
    return nil
}